walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,84 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> result = new ArrayList<TimeRange>(); 
+
+    long meetingRequestDuration = request.getDuration();
+    Collection<Event> eventsWithRequestedAttendees = getEventsWithRequestedAttendees(events, request.getAttendees());
+    Collection<TimeRange> availabeRanges = new ArrayList<>();
+
+    if(events.size()==0 && meetingRequestDuration<=TimeRange.WHOLE_DAY.end()){ 
```

Comment:
There should be spaces between if and (. And it should be applied to the rest of the file also.

Reply:
I believed i fixed the space you were referring to. Sure we can go through that as well in our 1:1 but do you think that enough time to explain complexity. I also have to do the second part of the week 5 project. @rayliu0317